runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -369,7 +369,10 @@ public State call() throws Exception {
         } catch (IOException e) {
           State state = getState();
           if (state.isTerminal()) {
-            LOG.warn("Job is already terminated. State is {}", state);
+            LOG.warn("Cancel failed because job is already terminated. State is {}", state);
+            return state;
+          } else if (e.getMessage().contains("has terminated")) {
```

Comment:
Please add a comment quoting an example exact message that we're matching against.

Also, add a test?

Also, fix the failing test and run Dataflow ValidatesRunner :)